### DISCUSSION TITLE
[Feature Request] "All games" page

### DISCUSSION THREAD

**Discussion by redComicsar** [RetroAchievements/RAWeb#2642] [Ideas]
Please add sorting of games by "most players" on the "All games" page. And it's better that this page looks like a console pages. It would also be nice to have filters to exclude "Subsets", "Hacks", etc.

![1](https://github.com/user-attachments/assets/8c33d7ad-e675-4f88-8a45-673a8287697d)

![2](https://github.com/user-attachments/assets/995e51bb-e0ee-4274-9f45-f95582a46989)

**Comment by wescopeland** [comment]
Hi @redComicsar!

At the moment, we have some issues with how subsets, hacks, etc are stored in the database that make this functionality very difficult to implement. Long story short: if we implement this feature today, it will be very very very slow.

This is an ongoing cleanup effort but at the moment I have no ETA. Subset filtering I'd prefer to not support until we actually have multiset functionality available for players.

**Reply by redComicsar** [reply]
Perhaps a temporary solution could be to use the "Game kinds" filter, which is implemented on the "Beaten Games Leaderboard" page? Anyway, thanks for the quick response!

![3](https://github.com/user-attachments/assets/096843f7-ad83-4c75-9aef-709a55ca5ee5)

**Reply by wescopeland** [reply]
Sure thing! On this page we actually use a trick to get around the performance issues 😅

When you beat a game, if it's for a hack, we store it in a special row that logs specifically how many hacks you've beaten. Likewise for homebrew, prototype, etc. Then on the page, based on your filters, we do very basic sums from these special rows.

Originally the page did not precalculate any of this stuff, and each pageview took ~30 seconds even when using extremely fast server hardware.

Unfortunately that kind of trick doesn't work on an all games page. The only way to make the page fast is to store the data properly.

**Reply by redComicsar** [reply]
In that case, all we can do is wait. Can I ask two more small questions?

1) Can at least sorting by "Most players" be implemented in the current state? Or are these all parts of the same problem? If sorting can be implemented, then I will temporarily ignore the lines "Subsets", "Hacks", etc. Even this would be a great help in choosing a game.

2) Is it possible to make an exception in this and other lists for games where I have unlocked achievements?
